### ISSUE TITLE
DataStore conflict handle cannot handle delete mutation

### ISSUE THREAD

**Issue by HuiSF** [aws-amplify/amplify-android#2261]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Java, Kotlin

### Amplify Categories

DataStore

### Gradle script dependencies

<details>

```groovy
// Put output below this line


```

</details>

### Environment information

<details>

```
# Put output below this line


```

</details>

### Use case of DataStore conflict handler:

1. I'm deleting a local model
2. There is a conflict happened during the mutation
3. I want to make the local delete mutation always win, so I use `RETRY_LOCAL` strategy

### Issue
`RETRY_LOCAL` strategy doesn't make the local delete mutation work.

### Potential cause

The `publish` function used inside the conflict handler, seems always generate an update mutation on the resolved model, and not capable to generate a delete mutation following the use case described above.

https://github.com/aws-amplify/amplify-android/blob/9074d878460bd9e53506c08887515c5563950976/aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/ConflictResolver.java#L169-L176

### Code Snippet

```java
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by stevegaunt** [comment]
@HuiSF are there any updates with this?

**Comment by stevegaunt** [comment]
@HuiSF any idea when this bug will get resolved??  It's causing data inconsistency

**Comment by mattcreaser** [comment]
@stevegaunt could you share a little information about the reproduction steps you're seeing this issue with?

I will make a fix here but I'd like to ensure I am covering all the circumstances.

**Comment by stevegaunt** [comment]
@mattcreaser 

It comes down to timing mainly due to network inconsistencies with our users.. 
 
If a mutation is created or updated and synced to the remote database, but before amplify has received the response of the outgoing pending event to update the version locally,  we have already within the app done a delete mutation on that object..  however, it's version number is not updated to the synced version number DynamoDB will throw a conflict back to app sync and consequently to the conflict resolution function..   It looks like the received local and remote model instances doesn't include conflict detection meta fields such as _deleted 

simple step to reproduce would be
1. mutate create object and wait until it's synced.
2. manually increment  DynamoDB version on that object
3. mutate delete object  - should get conflict resolution for the deleted object.


for our scenarios as it stands at the moment, the local change will be the correct version.

In future, we may have different conflict resolution for models shared across different devices and users( but that will be different conflict management based on the model)

**Comment by mattcreaser** [comment]
This has been released in version `2.11.1`
